spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/tf/loaders/DataFlowOps.scala

```diff
@@ -0,0 +1,269 @@
+
+import java.nio.ByteOrder
+
+import com.intel.analytics.bigdl.nn.Identity
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+
+import com.intel.analytics.bigdl.Module
+import com.intel.analytics.bigdl.nn.ops._
+import com.intel.analytics.bigdl.tensor.TensorNumericMath
+import com.intel.analytics.bigdl.utils.tf.Context
+import com.intel.analytics.bigdl.utils.tf.loaders.Utils._
+import org.tensorflow.framework.{DataType, NodeDef}
+
+import scala.reflect.ClassTag
+
+private[bigdl] class TensorArrayV3 extends TensorflowOpsLoader {
+
+  override def build[T: ClassManifest](
+    nodeDef: NodeDef,
+    byteOrder: ByteOrder,
+    context: Context[T]
+  )(implicit ev: TensorNumericMath.TensorNumeric[T]): Module[T] = {
+    val dynamicSize = getBoolean(nodeDef, "dynamic_size")
+    val clearAfterRead = getBoolean(nodeDef, "clear_after_read")
+    val identicalElementShapes = if (nodeDef.containsAttr("identical_element_shapes")) {
+      getBoolean(nodeDef, "identical_element_shapes")
+    } else {
+      false
+    }
+    val tensorArrayName = getString(nodeDef, "tensor_array_name")
+
+    val t = getType(nodeDef, "dtype")
+    if (t == DataType.DT_FLOAT) {
+      new TensorArrayCreator[T, Float](
+        dynamicSize = dynamicSize,
+        clearAfterRead = clearAfterRead,
+        identicalElementShapes = identicalElementShapes,
+        tensorArrayName = if (tensorArrayName == "") null else tensorArrayName
+      )
+    } else if (t == DataType.DT_DOUBLE) {
+      new TensorArrayCreator[T, Double](
+        dynamicSize = dynamicSize,
+        clearAfterRead = clearAfterRead,
+        identicalElementShapes = identicalElementShapes,
+        tensorArrayName = if (tensorArrayName == "") null else tensorArrayName
+      )
+    } else if (t == DataType.DT_INT32) {
+      new TensorArrayCreator[T, Int](
+        dynamicSize = dynamicSize,
+        clearAfterRead = clearAfterRead,
+        identicalElementShapes = identicalElementShapes,
+        tensorArrayName = if (tensorArrayName == "") null else tensorArrayName
+      )
+    } else {
+      throw new UnsupportedOperationException(s"Not support load TensorArrayV3 with data type $t")
+    }
+  }
+}
+
+private[bigdl] class TensorArrayGradV3 extends TensorflowOpsLoader {
+
+  override def build[T: ClassManifest](
+    nodeDef: NodeDef,
+    byteOrder: ByteOrder,
+    context: Context[T]
+  )(implicit ev: TensorNumericMath.TensorNumeric[T]): Module[T] = {
+    val source = getString(nodeDef, "source")
+    new TensorArrayGrad[T](source)
+  }
+}
+
+class TensorArrayGatherV3 extends TensorflowOpsLoader {
+  override def build[T: ClassManifest](
+    nodeDef: NodeDef,
+    byteOrder: ByteOrder,
+    context: Context[T]
+  )(implicit ev: TensorNumericMath.TensorNumeric[T]): Module[T] = {
+    val t = getType(nodeDef, "dtype")
+    if (t == DataType.DT_FLOAT) {
+      new TensorArrayGather[T, Float]()
+    } else if (t == DataType.DT_DOUBLE) {
+      new TensorArrayGather[T, Float]()
```

Comment:
Is this a typo? Should be `TensorArrayGather[T, Double]()`?

Reply:
fixed